src/components/FlawAffects/FlawAffects.vue

```diff
@@ -0,0 +1,757 @@
+  } else {
+    selectedModules.value.push(moduleName);
+  }
+}
+
+function moduleBtnTooltip(moduleName: string) {
+  return moduleName
+    + (moduleTrackersCount(moduleName) === 0
+      ? '\nThis module has no trackers associated'
+      : '');
+}
+
+function isBeingEdited(affect: ZodAffectType) {
+  return affectsBeingEdited.value.includes(affect);
+}
+
+function getAffectPriorEdit(affect: ZodAffectType): ZodAffectType {
+  return affectValuesPriorEdit.value.find(prior => prior.uuid && prior.uuid === affect.uuid) || affect;
+}
+
+function editAffect(affect: ZodAffectType) {
+  if (isAffectSelected(affect)) {
+    toggleAffectSelection(affect);
+  }
+  if (!isBeingEdited(affect)) {
+    affectsBeingEdited.value.push(affect);
+    const affectCopy = clone(affect) as ZodAffectType;
+    affectValuesPriorEdit.value.push(affectCopy);
+  }
+}
+
+function editSelectedAffects() {
+  selectedAffects.value.forEach(editAffect);
+}
+
+function commitChanges(affect: ZodAffectType) {
+  const matchAffect = affectsMatcherFor(affect);
+  const updateIndex = flaw.value.affects.findIndex(matchAffect);
+  flaw.value.affects.splice(updateIndex, 1, affect);
+  resetStagedAffectEdit(affect);
+}
+
+function cancelChanges(affect: ZodAffectType) {
+  resetStagedAffectEdit(affect);
+}
+
+function resetStagedAffectEdit(affect: ZodAffectType) {
+  const matchAffect = affectsMatcherFor(affect);
+  const editingIndex = affectsBeingEdited.value.findIndex(matchAffect);
+  affectsBeingEdited.value.splice(editingIndex, 1);
+  affectValuesPriorEdit.value.splice(affectValuesPriorEdit.value.findIndex(matchAffect), 1);
+  if (isAffectSelected(affect)) {
+    toggleAffectSelection(affect);
+  }
+}
+
+function commitAllChanges() {
+  const affectsToCommit = [...affectsBeingEdited.value];
+  affectsToCommit.forEach(commitChanges);
+}
+
+function cancelAllChanges() {
+  const affectsToCancel = [...affectsBeingEdited.value];
+  affectsToCancel.forEach(cancelChanges);
+}
+
+function revertAffectToLastSaved(affect: ZodAffectType) {
+  const saved = initialAffects.value.find(a => a.uuid === affect.uuid);
+  const index = flaw.value.affects.findIndex(a => a.uuid === affect.uuid);
+  if (index !== -1 && saved) {
+    flaw.value.affects[index] = { ...saved };
+  }
+  if (isAffectSelected(affect)) {
+    toggleAffectSelection(affect);
+  }
+}
+
+function revertAllAffects() {
+  const affectsToRestore = [...modifiedAffects.value];
+  affectsToRestore.forEach((affect) => {
+    if (!affectsBeingEdited.value.includes(affect)) {
+      revertAffectToLastSaved(affect);
+    }
+  });
+}
+
+// Create affects
+const newAffects = computed(() => flaw.value.affects.filter(affect => !affect.uuid));
+
+function addNewAffect() {
+  addAffect({
+    embargoed: props.embargoed,
+    affectedness: 'NEW',
+    resolution: '',
+    delegated_resolution: '',
+    ps_module: `NewModule-${newAffects.value.length}`,
+    ps_component: `NewComponent-${newAffects.value.length}`,
```

Comment:
Just being picky here, but probably this is better for the users?

```suggestion
    ps_module: `NewModule-${newAffects.value.length + 1}`,
    ps_component: `NewComponent-${newAffects.value.length + 1}`,
```

Reply:
Agreed